pkg/cluster/delete.go

```diff
@@ -377,6 +378,77 @@ func (m *manager) deleteClusterMsiCertificate(ctx context.Context) error {
 	return err
 }
 
+func (m *manager) deleteIdentityFederations(ctx context.Context) error {
+	if m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile == nil {
+		return nil
+	}
+
+	if m.clusterMsiFederatedIdentityCredentials == nil {
+		m.log.Warning("cluster MSI federated identity credentials client is nil, trying to initialize")
+		err := m.initializeClusterMsiClients(ctx)
+		if err != nil {
+			m.log.Errorf("cluster MSI federated identity credentials client initialization failed with error: %v", err)
+			return nil
```

Comment:
Why are returning nil here? I'd expect this to return `err`, and whoever is calling `deleteIdentityFederations` can decide what to do with that error. Example: log the error with "unable to delete due to client init failure, will continue...".

Reply:
As per the previous discussions [here](https://github.com/Azure/ARO-RP/pull/3847#discussion_r1779260847) we don't want to block the resource cleanup process if we aren't able to delete the Fed cred Identity. As of now, we return `error` when we get error parsing a resource ID of the Identity. 

Today this function is called only during resource cleanup from `Delete()` func under `delete.go`

Reply:
We could encounter an error here for one of two reasons:

1. The cx deleted their cluster MSI; we definitely want to log the error and return nil in this case
2. There's some other RP-side error; we probably want to error out in this case